lib/internal/modules/cjs/loader.js

```diff
@@ -1246,6 +1245,23 @@ Module._extensions['.node'] = function(module, filename) {
   return process.dlopen(module, path.toNamespacedPath(filename));
 };
 
+Module._extensions['.mjs'] = function(module, filename) {
+  const ESMLoader = asyncESM.ESMLoader;
+  const url = `${pathToFileURL(filename)}`;
+  const job = ESMLoader.getModuleJobWorker(url, 'module');
+  /* So long as builtin esm resolve is sync, this will complete sync
+   * (if the loader is extensible, an async loader will be have an observable
+   * effect here)
+   */
+  const instantiated = promiseWait(job.instantiate());
+  module.exports = instantiated.getNamespace();
+  /**
+   * So long as the module doesn't contain TLA, this will be sync, otherwise it
+   * appears async
+   */
+  promiseWait(instantiated.evaluate(-1, false));
```

Comment:
I believe this will throw if the last expression of a module source returns a rejected promise.

Reply:
Oooo spicy - is there a way to distinguish that from a TLA module that threw once https://github.com/nodejs/node/pull/30370 is in?

Reply:
once TLA is enabled, the result of `module.evaluate()` is either `undefined` or `Promise<undefined>`.

Reply:
Hah, yeah, I guess I just don't need this `promiseWait` call until #30370 is in. I'll remove it for now.

Reply:
> once TLA is enabled, the result of module.evaluate() is either undefined or Promise<undefined>.

Suppose I have
```
// @filename: mod.mjs
new Promise(resolve => resolve(undefined))
```
how am I to distinguish that from a module with TLA? I think conflating the evaluation of the final expression and the completion promise, API-wise, is problematic.